documentation/debugging.rst

```diff
@@ -0,0 +1,104 @@
+Debugging
+##########
+
+This section is to talk about examples of how you would debug your config.
+
+---------------
+Basic knowledge
+---------------
+
+Sometimes, things aren't as point and click as you might expect it to be.  
+Here are some helpful tips that might help you on your way to making bindings.
+
+
+---------------
+inconsistencies
+---------------
+
+Binder moves down your ``all_includes_file`` sequentially, sometimes you might end up with errors,
+and sometimes you might not.  This is almost always caused by your ``all_includes_file`` being
+different between runs.  The order shouldn't really matter, but nail it down to at least be
+consistent, and then move on to the next step.
+
+--------------
+Build failures
+--------------
+
+Sometimes you might get an error when building your compiled module.  binder has trouble  
+when standard libraries include eachother.  We have already mapped many of these
+interactions in ``types.cc``, but we still might be missing some internal knowledge about
+all of the c++ std library modules out there.  An error like this almost always
+manifests itself with an abundance of errors complaining about ``std::bits`` or ``bits``.
```

Comment:
Hmm... unless i am mistaken code is not in `std::bits` but rather in directory `bits/bits_<some_std_include>`